src/txn/txn.c

```diff
@@ -709,45 +718,42 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 		 * than stable timestamp.
 		 */
 		WT_ERR(__wt_timestamp_validate(
-		    session, "commit", ts, &cval, false));
+		    session, "commit", ts, &cval, !prepare));
 		txn->commit_timestamp = ts;
 		__wt_txn_set_commit_timestamp(session);
+		if (!prepare)
+			txn->durable_timestamp = txn->commit_timestamp;
 	}
 
-	prepare = F_ISSET(txn, WT_TXN_PREPARE);
 	if (prepare && !F_ISSET(txn, WT_TXN_HAS_TS_COMMIT))
 		WT_ERR_MSG(session, EINVAL,
 		    "commit_timestamp is required for a prepared transaction");
 
-	/* Durable timestamp is required for a prepared transaction. */
-	if (prepare) {
-		WT_ERR(__wt_config_gets_def(
-		    session, cfg, "durable_timestamp", 0, &cval));
-		if (cval.len != 0) {
-			WT_ERR(__wt_txn_parse_timestamp(
-			    session, "durable", &ts, &cval));
-			WT_ERR(__wt_timestamp_validate(
-			    session, "durable", ts, &cval, true));
-			txn->durable_timestamp = ts;
-		} else
-			/*
-			 * If durable timestamp is not given, commit timestamp
-			 * will be considered as durable timestamp.
-			 * TODO : error if durable timestamp is not given.
-			 */
-			txn->durable_timestamp = txn->commit_timestamp;
-
-	} else
-		txn->durable_timestamp = txn->commit_timestamp;
+	/*
+	 * Durable timestamp is required for a prepared transaction.
+	 * If durable timestamp is not given, commit timestamp will be
+	 * considered as durable timestamp. We don't flag error if durable
+	 * timestamp is not specified for prepared transactions, but will flag
+	 * error if durable timestamp is specified for non-prepared
+	 * transactions.
+	 */
+	WT_ERR(__wt_config_gets_def(
+	    session, cfg, "durable_timestamp", 0, &cval));
+	if (cval.len != 0) {
+		if (!prepare)
+			WT_ERR_MSG(session, EINVAL,
+			    "durable_timestamp should not be given for "
+			    "non-prepared transaction");
```

Comment:
This part of the change seems unrelated to implementing the `assert` functionality. Is it just a small bug fix or should it be part of a differrent ticket?

Reply:
Yes, this part is not part of assert functionality, it is more of error checking if the user has given durable timestamp for a non-prepared transaction.

Reply:
That is fine that it is a small fix you noticed and added while you were working in this area of code. I just wanted to make sure I wasn't misunderstanding what I was reading.